scripts/validate_unwanted_patterns.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python
+"""
+Unwanted patterns test cases.
+"""
+
+import argparse
+import os
+import sys
+import token
+import tokenize
+from typing import Callable, Generator, List, Tuple
+
+FILE_EXTENSIONS_TO_CHECK = (".py", ".pyx", ".pyx.ini", ".pxd")
+
+
+def main(
+    function: Callable[[str], Generator[Tuple[str, int, str], None, None]],
+    source_path: str,
+    output_format: str,
+) -> bool:
+    """
+    Main entry point of the script.
+
+    Parameters
+    ----------
+    function : Callable
+        Function to execute for the test case.
+    source_path : str
+        Source path representing path to a file/directory.
+    output_format : str
+        Output format of the error message.
+
+    Returns
+    -------
+    bool
+        True if found any patterns are found related to the given function.
+
+    Raises
+    ------
+    ValueError
+        If the `source_path` is not pointing to existing file/directory.
+    """
+    if not os.path.exists(source_path):
+        raise ValueError(
+            "Please enter a valid path, pointing to a valid file/directory."
+        )
+
+    is_failed: bool = False
+
+    if os.path.isfile(source_path):
+        for source_path, line_number, msg in function(source_path):
+            is_failed = True
+            print(
+                output_format.format(
+                    source_path=source_path, line_number=line_number, msg=msg
+                )
+            )
+
+    for subdir, _, files in os.walk(source_path):
+        for file_name in files:
+            if any(
+                file_name.endswith(extension) for extension in FILE_EXTENSIONS_TO_CHECK
+            ):
+                for source_path, line_number, msg in function(
+                    os.path.join(subdir, file_name)
+                ):
+                    is_failed = True
+                    print(
+                        output_format.format(
+                            source_path=source_path, line_number=line_number, msg=msg
+                        )
+                    )
+    return is_failed
+
+
+def STC(source_path: str) -> Generator[Tuple[str, int, str], None, None]:
```

Comment:
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Can we rename the methods to something more readable? Thanks!

Reply:
Yes, you're absolutely right